lib/ansible/modules/remote_management/ucs/ucs_macpool.py

```diff
@@ -0,0 +1,201 @@
+    argument_spec = ucs_argument_spec
+    argument_spec.update(_argument_mo())
+    argument_spec.update(_argument_custom())
+    argument_spec.update(_argument_connection())
+
+    return AnsibleModule(argument_spec,
+                         supports_check_mode=True)
+
+
+def _get_mo_params(params):
+    args = {}
+    for key in _argument_mo():
+        if params.get(key) is None:
+            continue
+        args[key] = params.get(key)
+    return args
+
+
+def setup_macpool(login_handle, module):
+    from ucsmsdk.mometa.macpool.MacpoolPool import MacpoolPool
+    from ucsmsdk.mometa.macpool.MacpoolBlock import MacpoolBlock
+
+    ansible = module.params
+    args_mo = _get_mo_params(ansible)
+
+    changed = False
+
+    for mac in args_mo['mac_list']:
+        exists = False
+        dn = args_mo['org_dn'] + '/mac-pool-' + mac['name']
+        mo = login_handle.query_dn(dn)
+        if mo:
+            # check top-level mo props
+            kwargs = {}
+            kwargs['assignment_order'] = mac['order']
+            if (mo.check_prop_match(**kwargs)):
+                # top-level props match, check next level props
+                if(mac['to'] != '' and mac['from'] != ''):
+                    block_dn = dn + '/block-' + mac['from'] + '-' + mac['to']
+                    mo_1 = login_handle.query_dn(block_dn)
+                    if mo_1:
+                        exists = True
+                else:
+                    exists = True
+
+        if ansible['state'] == 'absent':
+            if exists:
+                changed = True
+                if not module.check_mode:
+                    login_handle.remove_mo(mo)
+                    login_handle.commit()
+        else:
+            if not exists:
+                changed = True
+                if not module.check_mode:
+                    # create if mo does not already exist
+                    if 'description' not in mac:
+                        mac['description'] = ''
+                    mo = MacpoolPool(parent_mo_or_dn=args_mo['org_dn'],
+                                     name=mac['name'],
+                                     descr=mac['description'],
+                                     assignment_order=mac['order'])
+
+                    if(mac['to'] != '' and mac['from'] != ''):
+                        mo_1 = MacpoolBlock(parent_mo_or_dn=mo,
+                                            to=mac['to'],
+                                            r_from=mac['from'])
+                    login_handle.add_mo(mo, True)
+                    login_handle.commit()
+
+    return changed
+
+
+def setup(login_handle, module):
+    result = {}
+    err = False
+
+    try:
+        result['changed'] = setup_macpool(login_handle, module)
+    except Exception as e:
+        err = True
+        result['msg'] = "setup error: %s " % str(e)
+        result['changed'] = False
+
+    return result, err
+
+
+def main():
+    module = _ansible_module_create()
```

Comment:
I think this is confusing as it creates unneeded abstractions/indirections. I would keep this as simple as possible.

Why not just add parameters right here, and maybe comment what is MO-related and what is customized ?
It will make it much more clear what options are listed and helps keeping the documentation in sync as well.

*(This is just my opinion, if you disagree, that's fine too)*